From: Ray Strode <rstrode@redhat.com>
Date: Wed, 2 Mar 2022 16:31:24 -0500
Subject: data: Be a little more specific when matching PCI cards

udev sends a variety of events about devices during boot. Right
now we may be disabling wayland multiple times on Hi1710 ARM
machines, because we just match the device by id, and fail
to look at the event itself.

This commit just adds subsystem and action guards to specify
when the wayland disabling code should run a little more specifically.

(cherry picked from commit 280ce4fe1b58057958f312efa429d47faf556e94)
---
 data/61-gdm.rules.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/data/61-gdm.rules.in b/data/61-gdm.rules.in
index e9c7269..0f6e446 100644
--- a/data/61-gdm.rules.in
+++ b/data/61-gdm.rules.in
@@ -1,6 +1,12 @@
+
+SUBSYSTEM!="pci", GOTO="gdm_pci_device_end"
+ACTION!="bind", GOTO="gdm_pci_device_end"
+
 # disable Wayland on Hi1710 chipsets
 ATTR{vendor}=="0x19e5", ATTR{device}=="0x1711", GOTO="gdm_disable_wayland"
 
+LABEL="gdm_pci_device_end"
+
 # disable Wayland if modesetting is disabled
 KERNEL!="card[0-9]*", GOTO="gdm_nomodeset_end"
 SUBSYSTEM!="drm", GOTO="gdm_nomodeset_end"
